### ISSUE TITLE
functions will not be highlighted in cpp, and it seems confilcting with vim-cpp-enhanced-highlight

### ISSUE THREAD

**Issue by zhangwbnn** [ErichDonGubler/vim-sublime-monokai#3]
![image](https://user-images.githubusercontent.com/14917179/43582387-dc5f81e2-968e-11e8-8d6c-a94274b3adc6.png)
as the image show, function will not be highlighed, which is default for vim syntax. vim-cpp-enhanced-highlight can highlight functions for monokai, but can not do this for vim-sublime-monokai, would you fix this?

**Comment by ErichDonGubler** [comment]
Hey there! Thank you so much for taking the time to file an issue -- I appreciate the effort because it helps me make something better for everyone,  myself included!

I actually have been aware of this particular quirk with C++ highlight and use `vim-cpp-enhanced-highlight` myself. I worked on C++ highlighting before I got into the habit of annotating known deficiencies and their causes. I'll record what I remember here and then put it into the repo like the other deficiency notes I have.

---

**Choosing the `Normal` highlight for functions was actually purposeful and related to how Sublime highlights C++.** One of my favorite parts about most Sublime syntaxes is that they distinguish between function *declarations/definitions* and function *calls*. Unfortunately, the vanilla Vim highlighting groups use `cCustomFunc` for both, and so there's no way to distinguish between the two like Sublime would. Some people prefer to have `Normal` highlighting everywhere if they can't be differentiated, and others still prefer to highlight both the same way with something else (which seems to be the case with `vim-cpp-enhanced-highlight`. If you prefer the latter, then you should easily be able to add this to your `.vimrc`:

```viml
hi! link cCustomFunc Tag
```

Note that if I could have my cake and eat it too, I'd highlight the definitions/declarations with `Tag` (linked to `SublimeGreen`) and the calls to `Keyword` (linked to `SublimeAqua`) -- this would be total parity with Sublime's functionality and the objective of this particular color scheme.

Long story short, I've yet to find a solution I'm happy with here. The ideal would be to find a plugin that revamps the C++ highlighting enough to distinguish these two. There's several places I'm not quite happy with vanilla C++ highlighting, so if there's a plugin like that out there, suggestions would be welcome!

**Comment by ErichDonGubler** [comment]
@zhangwbnn: Just wanted to ping you here to get your feedback. I'll close this issue soon if I don't get a response from you, and I definitely want to make sure you have a chance to voice your opinion!

I consider the changes I've made noting the deficiencies with the C++ configuration sufficient, since I would expect most downstream users to go there first when they have questions. How do you feel about the workaround I suggested? I have included it in the deficiency notes, if you look at the commit referenced above.

**Comment by zhangwbnn** [comment]
hi ErichDonGubler, I am sorry that I was too busy to give you feedback. I took your suggestion to use cCustomFunc. It works as you said. I am using vim-sublime-monokai now and will pay attention to vim-sublime-monokai.

**Comment by ErichDonGubler** [comment]
Does that mean your issue is resolved? If so, yay!

**Comment by ErichDonGubler** [comment]
I'll close this issue, since I haven't heard any objections in almost two weeks. Thanks for reporting this -- feel free to reopen an issue if you're not happy with this resolution. :)
